tt_metal/impl/device/device.cpp

```diff
@@ -43,6 +44,112 @@ Device::Device(
     this->initialize(num_hw_cqs, l1_small_size, trace_region_size, l1_bank_remap, minimal);
 }
 
+/* Get all dispatch cores associated with this device. On return, my_dispatch_cores contains dispatch cores used by
+ * this device (split between cores on this device itself and if this is a remote device, the mmio device dispatch
+ * cores being used by this device). On return, other_dispatch_cores contains dispatch cores on this device that are
+ * used by other (remote) devices.
+*/
+void Device::get_associated_dispatch_phys_cores(
```

Comment:
We might want to pre-assign dispatch/prefetch cores (in a yaml or something). Discussing with Austin offline, we probably want to place all these FD2 cores for NOC0/1 to get highest perf.

Reply:
Yeah I just pulled this into a function because it's needed in multiple places now. Would be glad to see a better solution for assigning dispatch cores like Allan mentioned.

Reply:
This particular code is only querying already allocated cores. The info is used later to check for proper tear down of dispatch kernels.
we cannot use yaml since dispatch cores can be in different rows depending on harvesting on the chip.
chips in the cluster can have different harvest recipe.